### PR TITLE
Tests for stop points addition

### PR DESCRIPTION
**Pull request by mbounabi**

### CODE REVIEW

File: artemis/tests/fixtures/trip_delete_new_stop_point_9580_tgv_in_the_middle.json [hunk 1]
```diff
@@ -0,0 +1,785 @@
+{
```
Comment (pbougue) [inline artemis/tests/fixtures/trip_delete_new_stop_point_9580_tgv_in_the_middle.json]: This fixture (and some other) is unused so far.

File: readme.md [hunk 2]
```diff
@@ -2,7 +2,10 @@
 Artemis
 *******
 
-Business tests for navitia
+Business tests for Navitia
+
+A new version of Artemis is now available. It allows to run Artemis to run locally (or on any machine) by running services in docker containers.
```
Comment (mbounabi) [inline readme.md]: Note to myself: remove a 'to run'

File: artemis/readme.md [hunk 3]
```diff
@@ -1,36 +1,34 @@
...
-    - set URL_TYR
-    - set DATA_DIR (artemis data)
-    - set REFERENCE_FILE_PATH(the path to artemis references)
-    - set RESPONSE_FILE_PATH (Optional, output of the responses generated by artemis)
-    - set KIRIN_API (if needed)
+* In default_settings.py, set the following parameters:
+    - URL_JORMUN (with port)
+    - URL_TYR
+    - DATA_DIR (artemis data)
+    - REFERENCE_FILE_PATH(the path to artemis references)
```
Comment (pbougue) [inline artemis/readme.md]: minor: missing space before ` (`

File: artemis/test_mechanism.py [hunk 4]
```diff
@@ -377,9 +371,8 @@ def _api_call(self, url, response_checker):
             return
 
         response, url, _ = utils.request(url)
-
         filtered_response = response_checker.filter(response)
-
+
```
Comment (pbougue) [inline artemis/test_mechanism.py]: minor: trailing space

File: artemis/readme.md [hunk 5]
```diff
@@ -1,36 +1,34 @@
...
+    You can clone the repository navitia-docker-compose, available [here](https://github.com/CanalTP/navitia-docker-compose) and follow the instructions in the [readme](https://github.com/CanalTP/navitia-docker-compose/blob/master/README.md) file to have the whole Navitia environment running.
+    - Once you set up all correctly, launch this command at the root directory of the repo:
+    `docker-compose -f docker-compose.yml -f artemis/docker-artemis-instance.yml up`
+    - To run real-time tests, Kirin is needed. Please refer to these [instructions](https://github.com/CanalTP/navitia-docker-compose/blob/master/kirin/README.md) to have it in the docker-compose. In this case, the command is:
+    `docker-compose -f docker-compose.yml -f artemis/docker-artemis-instance.yml -f kirin/docker-compose_kirin.yml up`
 
-  For python2:
-  `pip install -r requirements_py2.txt`
+* Create a virtual environment with the python packages in requirements.txt (for python 3):
+    `mkvirtualenv --python=/usr/bin/python3 <venv_name>`
```
Comment (nonifier) [inline artemis/readme.md]: Or shorter:
 ```sh
mkvirtualenv -p python3 -r requirements.txt <venv_name>
```
 #OneLiner

File: artemis/readme.md [hunk 6]
```diff
@@ -1,36 +1,34 @@
...
+    - DATA_DIR (artemis data)
+    - REFERENCE_FILE_PATH (the path to artemis references)
+    - RESPONSE_FILE_PATH (Optional, output of the responses generated by artemis)
+    - KIRIN_API (if needed)
 
 * In the settings file, set 'USE_ARTEMIS_NG' to True
 
 * Launch tests: `py.test artemis/tests`
     - If you don't want to call `cities`, add `--skip_cities`
+    - If the data has already been binarized, add `--skip_bina`
```
Comment (nonifier) [inline artemis/readme.md]: :notes: :microphone: skiiiiiiiiiiiiiip bina bina :notes: :smiley:

File: artemis/utils.py [hunk 7]
```diff
@@ -143,6 +148,33 @@ def filter_dict(response, mask):
     return flask_restful.marshal(response, mask)
 
 
+def get_calling_test_function():
+    """
+    return the calling test method.
+
+    go back up the stack until a method with test in the name
```
Comment (nonifier) [inline artemis/utils.py]:
```suggestion
    go back up the stack until we find a method with the test's name
```

Comment (mbounabi) [reply artemis/utils.py]: I don't like to involve anyone in the comment, so I propose:
go back up the stack until a method with the test's name is found

File: artemis/readme.md (hunk 6 above)
Comment (mbounabi) [reply artemis/readme.md]: :musical_note: fo' real!

File: artemis/base_pytest.py [hunk 8]
```diff
@@ -318,7 +300,7 @@ def print_diff():
     ### Get the reference
 
     # Create the file name
-    filename = self.get_file_name()
+    filename = self._get_file_name()
```
Comment (pbougue) [inline artemis/base_pytest.py]: weird to have a private method declared outside

**Pull request merged**